### PR TITLE
Converted some packets to structs

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: -

* **Server Mode**: -

* **Description of Pull Request**:

### CODE REVIEW

File: src/map/clif.cpp [hunk 1]
```diff
@@ -6872,19 +6847,18 @@ void clif_heal(int fd,int type,int val) {
 /// 0148 <id>.L <type>.W
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:6847]:
```suggestion
/// 0148 <id>.L <type>.W (ZC_RESURRECTION)
```
And please remove it above (cant suggest via github...)

File: src/map/clif.cpp [hunk 2]
```diff
@@ -6872,19 +6847,18 @@ void clif_heal(int fd,int type,int val) {
...
 
-	nullpo_retv(bl);
+	packet.packetType = HEADER_ZC_RESURRECTION;
+	packet.gid = bl.id;
+	packet.type = 0;
 
-	WBUFW(buf,0)=0x148;
-	WBUFL(buf,2)=bl->id;
-	WBUFW(buf,6)=0;
+	clif_send( &packet, sizeof( packet ), &bl, (type == 1 ? AREA : AREA_WOS) );
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:6857]: There is not a single call with type != 1, consider removing that parameter.

File: src/map/clif.cpp [hunk 3]
```diff
@@ -6986,22 +6956,22 @@ void clif_map_property_mapall(int map_idx, enum map_property property)
...
-void clif_refine(int fd, int fail, int index, int val)
-{
-	WFIFOHEAD(fd,packet_len(0x188));
-	WFIFOW(fd,0)=0x188;
-	WFIFOW(fd,2)=fail;
-	WFIFOW(fd,4)=index+2;
-	WFIFOW(fd,6)=val;
-	WFIFOSET(fd,packet_len(0x188));
-}
+void clif_refine( int fd, uint16 result, int32 index, int8 val ){
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:6965]: Please refactor this from fd to sd and please consider creating an enum for the result parameter.
Index should be (u)int16.
Check if val could be replaced by accessing the inventory information of the refined item.

File: src/map/clif.hpp [hunk 4]
```diff
@@ -664,7 +664,7 @@ void clif_dropitem( map_session_data& sd, int index, int amount );
 void clif_delitem( map_session_data& sd, int index, int amount, short reason );
 void clif_update_hp(map_session_data &sd);
 void clif_updatestatus( map_session_data& sd, _sp type );
-void clif_changestatus(map_session_data* sd,int type,int val);	//area
+void clif_changestatus( map_session_data& sd, int32 type, int32 val );	//area
```
Comment (Lemongrass3110) [inline src/map/clif.hpp:667]:
```suggestion
void clif_changestatus( map_session_data& sd, int32 type, int32 val );
```

File: src/map/clif.hpp [hunk 5]
```diff
@@ -962,7 +962,7 @@ void clif_feel_hate_reset(map_session_data *sd);
 void clif_hominfo(map_session_data *sd, struct homun_data *hd, int flag);
 void clif_homskillinfoblock( homun_data& hd );
 void clif_homskillup( homun_data& hd, uint16 skill_id );
-void clif_hom_food(map_session_data *sd,int foodid,int fail);	//[orn]
+void clif_hom_food( map_session_data& sd, int32 foodid, uint8 result );	//[orn]
```
Comment (Lemongrass3110) [inline src/map/clif.hpp:965]:
```suggestion
void clif_hom_food( map_session_data& sd, int32 foodid, uint8 result );
```

File: src/map/clif.cpp [hunk 6]
```diff
@@ -1984,16 +1984,14 @@ void clif_homskillup( homun_data& hd, uint16 skill_id ){
 /// result:
 ///     0 = failure
 ///     1 = success
-void clif_hom_food( map_session_data *sd, int foodid, int fail ){
-	nullpo_retv( sd );
-
-	struct PACKET_ZC_FEED_MER p;
+void clif_hom_food( map_session_data& sd, int32 foodid, uint8 result ){
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:1987]:
```suggestion
void clif_hom_food( map_session_data& sd, int32 foodid, bool success ){
```

File: src/map/clif.cpp [hunk 7]
```diff
@@ -6275,48 +6258,40 @@ void clif_skill_warppoint( map_session_data* sd, uint16 skill_id, uint16 skill_l
...
 ///     0 = "Saved location as a Memo Point for Warp skill." in color 0xFFFF00 (cyan)
 ///     1 = "Skill Level is not high enough." in color 0x0000FF (red)
 ///     2 = "You haven't learned Warp." in color 0x0000FF (red)
 ///
 /// @param sd Who receives the message
 /// @param type What message
-void clif_skill_memomessage(map_session_data* sd, int type)
-{
-	int fd;
+void clif_skill_memomessage( map_session_data& sd, uint8 type ){
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:6270]: Enum for type?

File: src/map/clif.cpp [hunk 8]
```diff
@@ -6275,48 +6258,40 @@ void clif_skill_warppoint( map_session_data* sd, uint16 skill_id, uint16 skill_l
...
 }
 
 
-/// Teleport message (ZC_NOTIFY_MAPINFO).
-/// 0189 <type>.W
+/// Teleport message.
+/// 0189 <type>.W (ZC_NOTIFY_MAPINFO)
 /// type:
 ///     0 = "Unable to Teleport in this area" in color 0xFFFF00 (cyan)
 ///     1 = "Saved point cannot be memorized." in color 0x0000FF (red)
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:6284]: Value 2 is not documented

File: src/map/clif.cpp [hunk 9]
```diff
@@ -6275,48 +6258,40 @@ void clif_skill_warppoint( map_session_data* sd, uint16 skill_id, uint16 skill_l
...
 ///
 /// @param sd Who receives the message
 /// @param type What message
-void clif_skill_teleportmessage(map_session_data *sd, int type)
-{
-	int fd;
+void clif_skill_teleportmessage( map_session_data& sd, int16 type ){
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:6288]: Enum for type?

File: src/map/clif.cpp [hunk 10]
```diff
@@ -8447,21 +8417,19 @@ void clif_pet_emotion(struct pet_data *pd,int param)
...
+/// Result of request to feed a pet.
+/// 01a3 <result>.B <name id>.W (ZC_FEED_PET)
 /// result:
 ///     0 = failure
 ///     1 = success
-void clif_pet_food( map_session_data *sd, int foodid,int fail ){
-	nullpo_retv( sd );
-
-	struct PACKET_ZC_FEED_PET p;
+void clif_pet_food( map_session_data& sd, int32 foodid, uint8 result ){
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:8425]:
```suggestion
void clif_pet_food( map_session_data& sd, int32 foodid, bool success ){
```

**Review by Lemongrass3110 — APPROVED** [review]

File: src/map/clif.cpp [hunk 11]
```diff
@@ -3852,29 +3850,25 @@ void clif_updatestatus( map_session_data& sd, enum _sp type ){
...
 
 
-/// Notifies client of a parameter change of an another player (ZC_PAR_CHANGE_USER).
-/// 01ab <account id>.L <var id>.W <value>.L
-void clif_changestatus(map_session_data* sd,int type,int val)
-{
-	unsigned char buf[12];
+/// Notifies client of a parameter change of an another player.
+/// 01ab <account id>.L <var id>.W <value>.L (ZC_PAR_CHANGE_USER)
+void clif_changestatus( map_session_data& sd, int32 type, int32 val ) {
```
[the remark this answers is not in the document]
Comment (Atemo) [reply src/map/clif.cpp]: Maybe it's not the best but I changed it now

File: src/map/clif.cpp [hunk 12]
```diff
@@ -3852,29 +3850,20 @@ void clif_updatestatus( map_session_data& sd, enum _sp type ){
...
-/// 01ab <account id>.L <var id>.W <value>.L
-void clif_changestatus(map_session_data* sd,int type,int val)
-{
-	unsigned char buf[12];
+/// Notifies client of a parameter change of an another player.
+/// 01ab <account id>.L <var id>.W <value>.L (ZC_PAR_CHANGE_USER)
+/// var id:
+///     SP_MANNER
+///     ?
+void clif_changemanner( map_session_data& sd, int32 val ) {
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:3858]:
```suggestion
void clif_changemanner( map_session_data& sd ) {
```

File: src/map/clif.cpp [hunk 13]
```diff
@@ -3852,29 +3850,20 @@ void clif_updatestatus( map_session_data& sd, enum _sp type ){
...
+	PACKET_ZC_PAR_CHANGE_USER packet{};
 
-	nullpo_retv(sd);
+	packet.packetType = HEADER_ZC_PAR_CHANGE_USER;
+	packet.gid = sd.bl.id;
+	packet.type = static_cast<decltype(packet.type)>(SP_MANNER);
+	packet.value = val;
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:3864]:
```suggestion
	packet.value = sd.status.manner;
```

File: src/map/clif.hpp [hunk 14]
```diff
@@ -664,7 +683,7 @@ void clif_dropitem( map_session_data& sd, int index, int amount );
 void clif_delitem( map_session_data& sd, int index, int amount, short reason );
 void clif_update_hp(map_session_data &sd);
 void clif_updatestatus( map_session_data& sd, _sp type );
-void clif_changestatus(map_session_data* sd,int type,int val);	//area
+void clif_changemanner( map_session_data& sd, int32 val );
```
Comment (Lemongrass3110) [inline src/map/clif.hpp:686]:
```suggestion
void clif_changemanner( map_session_data& sd );
```

File: src/map/pc.cpp [hunk 15]
```diff
@@ -10863,7 +10863,7 @@ bool pc_jobchange(map_session_data *sd,int job, char upper)
 	}
 
 	if(sd->status.manner < 0)
-		clif_changestatus(sd,SP_MANNER,sd->status.manner);
+		clif_changemanner( *sd, sd->status.manner );
```
Comment (Lemongrass3110) [inline src/map/pc.cpp:10866]:
```suggestion
		clif_changemanner( *sd );
```

File: src/map/status.cpp [hunk 16]
```diff
@@ -10895,7 +10895,7 @@ int status_change_start(struct block_list* src, struct block_list* bl,enum sc_ty
 			tick = 60000;
 			val1 = battle_config.manner_system; // Mute filters.
 			if (sd) {
-				clif_changestatus(sd,SP_MANNER,sd->status.manner);
+				clif_changemanner( *sd, sd->status.manner );
```
Comment (Lemongrass3110) [inline src/map/status.cpp:10898]:
```suggestion
				clif_changemanner( *sd );
```

File: src/map/status.cpp [hunk 17]
```diff
@@ -13399,7 +13399,7 @@ int status_change_end(struct block_list* bl, enum sc_type type, int tid)
 			if (sd && sd->status.manner < 0 && tid != INVALID_TIMER)
 				sd->status.manner = 0;
 			if (sd && tid == INVALID_TIMER) {
-				clif_changestatus(sd,SP_MANNER,sd->status.manner);
+				clif_changemanner( *sd, sd->status.manner );
```
Comment (Lemongrass3110) [inline src/map/status.cpp:13402]:
```suggestion
				clif_changemanner( *sd );
```

File: src/map/status.cpp [hunk 18]
```diff
@@ -14220,7 +14220,7 @@ TIMER_FUNC(status_change_timer){
 	case SC_NOCHAT:
 		if(sd) {
 			sd->status.manner++;
-			clif_changestatus(sd,SP_MANNER,sd->status.manner);
+			clif_changemanner( *sd, sd->status.manner );
```
Comment (Lemongrass3110) [inline src/map/status.cpp:14223]:
```suggestion
			clif_changemanner( *sd );
```

File: src/map/pc.cpp [hunk 19]
```diff
@@ -10863,7 +10863,7 @@ bool pc_jobchange(map_session_data *sd,int job, char upper)
 	}
 
 	if(sd->status.manner < 0)
```
Comment (Lemongrass3110) [inline src/map/pc.cpp:10865]:
```suggestion
	if(sd->status.manner < 0){
```

File: src/map/pc.cpp [hunk 20]
```diff
@@ -10863,7 +10863,7 @@ bool pc_jobchange(map_session_data *sd,int job, char upper)
...
-		clif_changestatus(sd,SP_MANNER,sd->status.manner);
+		clif_changemanner( *sd, sd->status.manner );
```
Comment (Lemongrass3110) [inline src/map/pc.cpp:10867]:
```suggestion
		clif_updatestatus(*sd,SP_MANNER);
	}
```

File: src/map/clif.cpp [hunk 21]
```diff
@@ -4062,104 +4051,100 @@ void clif_refreshlook(struct block_list *bl, int id, int type, int val, enum sen
...
-	clif_updatestatus(*sd, SP_AGI);
-	clif_updatestatus(*sd, SP_VIT);
-	clif_updatestatus(*sd, SP_INT);
-	clif_updatestatus(*sd, SP_DEX);
-	clif_updatestatus(*sd, SP_LUK);
+	packet.hitSuccessValue = sd.battle_status.hit;
+	packet.avoidSuccessValue = sd.battle_status.flee;
+	packet.plusAvoidSuccessValue = sd.battle_status.flee2 / 10;
+	packet.criticalSuccessValue = sd.battle_status.cri / 10;
+	packet.ASPD = sd.battle_status.amotion; // aspd
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:4092]:
```suggestion
	packet.ASPD = sd.battle_status.amotion;
```

File: src/map/clif.cpp [hunk 22]
```diff
@@ -4062,104 +4051,100 @@ void clif_refreshlook(struct block_list *bl, int id, int type, int val, enum sen
...
+	packet.plusASPD = 0;  // always 0 (plusASPD)
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:4093]:
```suggestion
	packet.plusASPD = 0;

**Review by Lemongrass3110 — APPROVED** [review]

File: src/map/clif.cpp [hunk 23]
```diff
@@ -6275,48 +6254,41 @@ void clif_skill_warppoint( map_session_data* sd, uint16 skill_id, uint16 skill_l
...
 
 
-/// Teleport message (ZC_NOTIFY_MAPINFO).
-/// 0189 <type>.W
+/// Teleport message.
+/// 0189 <type>.W (ZC_NOTIFY_MAPINFO)
 /// type:
 ///     0 = "Unable to Teleport in this area" in color 0xFFFF00 (cyan)
 ///     1 = "Saved point cannot be memorized." in color 0x0000FF (red)
+///     2 = ?
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:6281]:
```suggestion
///     2 = "This skill cannot be used in this area"
///     3 = "This item cannot be used in this area"
```

**Pull request merged**